src/boot/bootctl.c

```diff
@@ -1132,6 +1199,17 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_graceful = true;
                         break;
 
+                case ARG_MAKE_MACHINE_ID_DIRECTORY:
+                        if (streq(optarg, "auto"))
+                                arg_make_machine_id_directory = -1;  /* default */
+                        else {
+                                r = parse_boolean_argument("--make-machine-id-directory=", optarg, &b);
```

Comment:
No need to go through `b`, just do `parse_boolean_argument("--make-machine-id-directory=", optarg, &arg_make_…);`

Reply:
Not only is that an incompatible pointer (though, this being C, who cares, _Bool is just an unsigned integer type with funny conversions and enforced 0/1), how do you foresee this behaving right on big-endian?

Reply:
Oh, `arg_make_machine_id_directory` is an int. Never mind then.